### ISSUE TITLE
Unimplemented FunctionDecl error reported, when decompiling the bitcode file from anvill

### ISSUE THREAD

**Issue by Muqi-Zou** [lifting-bits/rellic#116]
When I try to decompile the bitcode generated from anvill for following source code (see test.c/test_clang/test_clang.bc/test_clang_opt.bc) in attachment):
```
#include <stdio.h>
#include <stdlib.h>

int test() {
	return 0;
}

int main(int argc, char **argv) {
	int state = test();
	if (!state) {
		printf("You're not even trying.\n");
	}
	else {
		printf("That's odd.\n");
	}
}
```
I received errors from anvill and rellic:
anvill:
```
E0303 11:43:19.525399 17883 MCToIRLifter.cpp:571] Could not decode instruction at 56a reachable from instruction 564 in function at 540
E0303 11:43:19.688983 17883 Analyze.cpp:950] TODO: Found byte address 5a0 that is mapped to memory but doesn't directly resolve to a function or variable
E0303 11:43:19.689030 17883 Analyze.cpp:979] Unexpected type of value   %47 = inttoptr i64 1860 to i8* by   %48 = call i32 (i8*, ...) @sub_520(i8* %47)
E0303 11:43:19.689052 17883 Analyze.cpp:979] Unexpected type of value   %41 = inttoptr i64 1866 to i8* by   %42 = call i32 (i8*, ...) @sub_520(i8* %41)
bc file generated 
```
rellic:
```
F0303 11:43:19.821843 17887 Z3ConvVisitor.cpp:322] Unimplemented FunctionDecl visitor11
*** Check failure stack trace: ***
    @     0x5590fa7fe76c  google::LogMessageFatal::~LogMessageFatal()
    @     0x5590f93e2330  rellic::Z3ConvVisitor::VisitFunctionDecl()
Aborted (core dumped)
```
The json file (see test.json) I used is generated from binary ninja. 
I report issue here rather than at anvill, because the same bitcode could be decompiled correctly by llvmcbe project (see test_clang_opt.cbe.c & test_clang.cbe.c).
The commands I used to decompile can be found in "anvill_rellic.py"

anvill-decompile-json-10.0 --version:
```
anvill-decompile-json-10.0 version unknown
Commit Hash: be24fd5e77eb11463f16eba9ba54a8c92fca0067
Commit Date: 2021-02-22 17:35:57 +0100
Last commit by: Alessandro Gario [5714290+alessandrogario@users.noreply.github.com]
Commit Subject: [CI: Use a single job to publish releases (#122)]

All changes were committed prior to building.
```

rellic-decomp-10.0 --version:
```
rellic-decomp-10.0 version unknown
Commit Hash: eea7a544d789999019bd16711f8cd8dde056e678
Commit Date: 2021-02-18 20:37:25 -0500
Last commit by: Alessandro Gario [5714290+alessandrogario@users.noreply.github.com]
Commit Subject: [CMake: Add packaging support (#96)]

Uncommitted changes were present during build.
Using LLVM 10.0.0
```
the log infos are also attached
[test.tar.gz](https://github.com/lifting-bits/rellic/files/6077802/test.tar.gz)

**Comment by pgoodman** [comment]
Closing because this looks like a duplicate of #94.

**Comment by Muqi-Zou** [comment]
Actually these two issues are different, it is about how bitcode is generated:

1.  In #94, the bitcode file is generated from clang-10 directly: `clang-10 -emit-llvm -c -O0 test.c`

2.  Here #116, the bitcode file is generated from anvill + binary, for example: 
```
clang-10 -O0 -fno-builtin -fPIE -pie -fno-stack-protector -Wall test.c  -o test_clang 
&& python3.8  -m anvill --bin_in test_clang --spec_out test.json 
&& anvill-decompile-json-10.0 --spec test.json --bc_out test.bc
```

Meanwhile, I have tried to use rellic to decompile the bitcode being constructed as same as what #94 did, and the rellic actually works pretty well on the source code I showed above!
@pgoodman
